pkg/spec/job.go

```diff
@@ -50,6 +50,10 @@ func (job *JobSpecMod) Fix() error {
 	if job.RestartPolicy == "" {
 		job.RestartPolicy = api_v1.RestartPolicyOnFailure
 	}
+
+	if job.ControllerFields.Appversion != "" {
+		job.ControllerFields.ObjectMeta.Annotations = addKeyValueToMap(appVersion, job.ControllerFields.Appversion, job.ControllerFields.ObjectMeta.Annotations)
+	}
```

Comment:
If you do https://github.com/kedgeproject/kedge/pull/454#discussion_r152549094 this is not needed!

Reply:
THis can't be moved as that comment suggested, there is information about Kedge file structures in context of that function